scripts/test.py

```diff
@@ -0,0 +1,146 @@
+
+
+def is_windows():
+    '''
+    Check if the system is Windows
+    '''
+    return 'windows' == platform.system().lower()
+
+
+BUILD_ROOT = os.path.abspath(os.path.join(
+    os.path.split(os.path.abspath(__file__))[0], '..'))
+ALL_TESTS = {
+    'gfxrecon_application_test': [],
+    'gfxrecon_decode_test': [],
+    'gfxrecon_encode_test': [],
+    'gfxrecon_format_test': [],
+    'gfxrecon_util_test': [],
+    'VkLayer_gfxreconstruct_test': [],
+}
+
+
+class TestError(Exception):
+    '''
+    Raised on a test error
+    '''
+
+
+def import_build_script():
+    '''
+    Import the build script - reuse build script constants
+    '''
+    sys.path.insert(0, os.path.join(BUILD_ROOT, 'scripts'))
+    build_script = importlib.import_module('build')
+    return build_script
+
+
+def parse_args(build_script):
+    '''
+    Parse command line arguments
+    '''
+    arg_parser = argparse.ArgumentParser(
+        description="gfxreconstruct build script",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    arg_parser.add_argument('--version', dest='version',
+                            action='version', version=str(build_script.VERSION))
+    arg_parser.add_argument(
+        '--test-exe', dest='test_exe', metavar='TEST_EXE',
+        action='store', default=None,
+        help='Test executable to run')
+    arg_parser.add_argument(
+        '--test-args', dest='test_args', metavar='TEST_ARGS', nargs='+',
+        action='store',
+        help='Test arguments passed to the test executable')
+    arg_parser.add_argument(
+        '--build-dir', dest='build_dir', metavar='BUILD_DIR',
+        action='store', default=None)
+    arg_parser.add_argument(
+        '-a', '--arch', dest='architecture',
+        metavar='ARCH', action='store', choices=build_script.ARCHITECTURES,
+        default=build_script.DEFAULT_ARCHITECTURE,
+        help='Build target architecture. Can be one of: {0}'.format(
+                ', '.join(build_script.ARCHITECTURES)))
+    arg_parser.add_argument(
+        '-c', '--config', dest='configuration',
+        metavar='CONFIG', action='store', choices=build_script.CONFIGURATIONS,
+        default=build_script.DEFAULT_CONFIGURATION,
+        help='Build target configuration. Can be one of: {0}'.format(
+            ', '.join(build_script.CONFIGURATIONS)))
+    return arg_parser.parse_args()
+
+
+def run_test(test_exe, test_args):
+    '''
+    Run a single test
+    '''
+    run_test_args = [test_exe]
+    if args.test_args is not None:
+        run_test_args.extend(args.test_args)
+    run_test_work_dir = os.path.split(test_exe)[0]
+    if '' == run_test_work_dir:
+        run_test_work_dir = os.getcwd()
+    try:
+        test_result = subprocess.run(run_test_args, cwd=run_test_work_dir)
+        if 0 != test_result.returncode:
+            raise TestError('Running test', ' '.join(run_test_args), 'failed')
+    except Exception:
+        raise TestError('Failed to run test', test_exe, ' '.join(test_args))
```

Comment:
As above, include the working directory and the returncode.  Also, to avoid losing information from the original exception, do something like:
```
except Exception as e:
    raise TestError('Failed to run "{}" in "{}", returncode={}'.format(
        run_test_args, run_test_work_dir, test_result.returncode
    )) from e
```